src/httpserver.cpp

```diff
@@ -639,6 +640,42 @@ HTTPRequest::RequestMethod HTTPRequest::GetRequestMethod() const
     }
 }
 
+std::string HTTPRequest::GetQueryParameter(const std::string& key, const std::string& default_val) const
+{
+    std::string uri = evhttp_request_get_uri(req);
+    auto result = GetQueryParameterFromUri(uri, key, default_val);
+
+    return result;
+}
+
+std::string GetQueryParameterFromUri(const std::string& uri, const std::string& key, const std::string& default_val)
+{
+    auto query_start = uri.rfind('?');
+    if (query_start != std::string::npos) {
+        // Capture everything after the '?' sign, i.e. the query string
+        auto query_string = uri.substr(query_start + 1, uri.size() - (query_start + 1));
+
+        // Parse the query string into a key-value queue and iterate over it
+        struct evkeyvalq params_q;
+        evhttp_parse_query_str(query_string.c_str(), &params_q);
+
+        struct evkeyval* param;
+        for (param = params_q.tqh_first; param; param = param->next.tqe_next) {
+            if (param->key == key) {
+                std::string result = param->value;
+                evhttp_clear_headers(&params_q);
+                return result;
+            }
+        }
+        evhttp_clear_headers(&params_q);
+    }
+
+    if (default_val != "")
```

Comment:
Maybe use std::optional because an empty string could be the default?

Reply:
Agreed and done, for both `GetQueryParameterFromUri` and `HTTPRequest:GetQueryParameter`.